rfcs/formats/communication/relay.md

```diff
@@ -8,7 +8,109 @@
+```
+NetworkQuery { 
+  policy: ["CarrierMSP", "SellerMSP"], 
+  address: "trade-logistics.com:9089/wtln/wtln-channel:wtlncc:GetBillOfLadingByPurchaseOrder:PO59133361", 
+  requesting_relay: "", 
+  requesting_network: "americantfn", 
+  certificate: "-----BEGIN CERTIFICATE-----\nMIIB2zCCAXegAwIBAgIQfrI6AbTtEhuJa9XzLy1L/TAUBggqhkjOPQQDAgYIKoZI\nzj0DAQcwNzELMAkGA1UEBhMCR0IxDzANBgNVBAcMBkxvbmRvbjEXMBUGA1UECgwO\nU2VsbGVyQmFua05vZGUwHhcNMjIwMzIyMDAwMDAwWhcNMjcwNTIwMDAwMDAwWjA3\nMQswCQYDVQQGEwJHQjEPMA0GA1UEBwwGTG9uZG9uMRcwFQYDVQQKDA5TZWxsZXJC\nYW5rTm9kZTAqMAUGAytlcAMhAMkMVTNq8lr1I4o4ivrPUNLoeV2Ab4U9PsReYo+w\n6nEio4GJMIGGMB0GA1UdDgQWBBQaqL67Z4elSpuWHjc0wfShLEBPdDAPBgNVHRMB\nAf8EBTADAQH/MAsGA1UdDwQEAwIChDATBgNVHSUEDDAKBggrBgEFBQcDAjAfBgNV\nHSMEGDAWgBSUmpz6AYRj0sgN5TPBwA1CSs34ZDARBgorBgEEAYOKYgEBBAMCAQYw\nFAYIKoZIzj0EAwIGCCqGSM49AwEHA0gAMEUCIQCnEpDO8MNL5vL5Uiig0+9iQHX4\nBLPJZXw591f3bB8xlQIgC5ieNFiV126RvjAmuq3R6bkl//3vECCcEOzDvK424y8=\n-----END CERTIFICATE-----", 
+  requestor_signature: "c+5UYaA4bZB7AI4msFdo0DTZFMnEskhlPWhB8/2IFSzwJmOopz980kxcuC/KjF3T8GQyjgMNqadaLEnKtZm3Aw==",
+  nonce: "f34bc09f-8ae1-439f-ab34-5414a289e1dd", 
+  requesting_org: "SellerBankNode",
+  confidential: false
+}
+```
+
+
+## GetStateMessage
+
+This payload is used by the client application to poll for the query response from the relay. It is the argument for [GetState](../../models/infrastructure/relays.md#api-for-application-client) API of the relay.
+
+```protobuf
+message GetStateMessage {
+  string request_id = 1;
+}
+```
+
+The `request_id` field should be the same id returned by the relay as part of [ACK](../views/request.md#ack) on the successful execution of [RequestState](../../models/infrastructure/relays.md#api-for-application-client) API. 
+
+## GRPC Service APIs
+
+```protobuf
+// definitions of all messages used in the datatransfer protocol
+service DataTransfer {
+  // the requesting relay sends a RequestState request to the remote relay with a
+  // query defining the data it wants to receive
+  rpc RequestState(common.query.Query) returns (common.ack.Ack) {}
+  // the remote relay asynchronously sends back the requested data with
+  // SendState
+  rpc SendState(common.state.ViewPayload) returns (common.ack.Ack) {}
+  // Handling state sent from the driver.
+  rpc SendDriverState(common.state.ViewPayload) returns (common.ack.Ack){}
+}
+```
+
+`RequestState` and `SendState` APIs are explained in more details [here](../../models/infrastructure/relays.md#api-for-other-relays), while `SendDriverState` API is explained [here](../../models/infrastructure/relays.md#api-for-driver).
+
+## Database
+
+Relay has a database that is used to store remote queries and state of the local queries at different stages of the [data sharing](../../protocols/data-sharing/generic.md) protocol. There are two instances of the database, one for local, and other for remote. The path to both databases are configurable. Currently relay implements only two methods for database:
```

Comment:
We should write this in the style of a recommendation. So start with "A relay should maintain a database to store remote....". Last sentence: "The relay database should offer the following two methods:". Also "There should be two instances", etc. (replace "are" with "should be").